Utilities/build-script-helper.py

```diff
@@ -68,13 +68,16 @@ def swiftpm_bin_path(swift_exec: str, swiftpm_args: List[str], additional_env: D
     return check_output(cmd, additional_env=additional_env, capture_stderr=False, verbose=verbose).strip()
 
 
-def get_build_target(swift_exec: str, args: argparse.Namespace) -> str:
+def get_build_target(swift_exec: str, args: argparse.Namespace, cross_compile: bool = False) -> str:
     """Returns the target-triple of the current machine or for cross-compilation."""
     try:
         command = [swift_exec, '-print-target-info']
+        if cross_compile:
+            cross_compile_json = json.load(open(args.cross_compile_config))
+            command += ['-target', cross_compile_json["target"]]
         target_info_json = subprocess.check_output(command, stderr=subprocess.PIPE, universal_newlines=True).strip()
         args.target_info = json.loads(target_info_json)
-        if platform.system() == 'Darwin':
+        if re.search('-apple-macosx', args.target_info["target"]["unversionedTriple"]):
```

Comment:
Do we need do do a full regex search here? Wouldn’t substring matching be sufficient?

```suggestion
        if '-apple-macosx' in args.target_info["target"]["unversionedTriple"]:
```

Reply:
No, I simply copied this over from [the existing check](https://github.com/apple/sourcekit-lsp/blob/main/Utilities/build-script-helper.py#L140), as I'm unfamiliar with the Python alternatives. I'll change them all to your suggestions.

Reply:
I wonder why they use regex matching. Does case sensitivity matter here? Python’s `in` is case-senstivive while `re.search` might be case-insensitive.